src/Chainweb/Pact/Service/PactInProcApi.hs

```diff
@@ -75,12 +84,53 @@ withPactService' ver cid logger memPoolAccess cdbv action = do
 maxBlockSize :: Int64
 maxBlockSize = 10000
 
-pactMemPoolAccess :: Logger logger => MempoolBackend ChainwebTransaction -> logger -> MemPoolAccess
-pactMemPoolAccess mempool theLogger _height _hash bHeader = do
-    let forkFunc = (mempoolProcessFork mempool) (logFunction theLogger) -- :: BlockHeader -> IO (V.Vector ChainwebTransaction)
+pactMemPoolAccess
+    :: Logger logger
+    => MempoolConsensus ChainwebTransaction
+    -> logger
+    -> MemPoolAccess
+pactMemPoolAccess mpc logger = MemPoolAccess
+    { mpaGetBlock = pactMemPoolGetBlock mpc logger
+    , mpaSetLastHeader = pactMempoolSetLastHeader mpc logger
+    , mpaProcessFork = pactProcessFork mpc logger
+    }
+
+pactMemPoolGetBlock
+    :: Logger logger
+    => MempoolConsensus ChainwebTransaction
+    -> logger
+    -> (BlockHeight -> BlockHash -> BlockHeader -> IO (Vector ChainwebTransaction))
+pactMemPoolGetBlock mpc theLogger height hash _bHeader = do
+    logFn theLogger Info $! "pactMemPoolAccess - getting new block of transactions for "
+        <> "height = " <> sshow height <> ", hash = " <> sshow hash
+    mempoolGetBlock (mpcMempool mpc) maxBlockSize
+  where
+   logFn :: Logger l => l -> LogFunctionText -- just for giving GHC some type hints
+   logFn = logFunction
+
+pactProcessFork
+    :: Logger logger
+    => MempoolConsensus ChainwebTransaction
+    -> logger
+    -> (BlockHeader -> IO ())
+pactProcessFork mpc theLogger bHeader = do
+    let forkFunc = (mpcProcessFork mpc) (logFunction theLogger)
     txHashes <- forkFunc bHeader
```

Comment:
Reintroduction could also happen directly inside `processFork`

Reply:
It could, I don't know if that is better or worse -- right now processFork is just calculating the list of transactions to reintroduce.